apps/prairielearn/src/pages/studentAssessmentInstance/studentAssessmentInstance.html.ts

```diff
@@ -0,0 +1,751 @@
+import { EncodedData } from '@prairielearn/browser-utils';
+import { html, unsafeHtml } from '@prairielearn/html';
+import { renderEjs } from '@prairielearn/html-ejs';
+
+import { Modal } from '../../components/Modal.html.js';
+import { StudentAccessRulesPopover } from '../../components/StudentAccessRulesPopover.html.js';
+import { compiledScriptTag } from '../../lib/assets.js';
+import { formatPoints } from '../../lib/format.js';
+
+export function StudentAssessmentInstance({ resLocals }: { resLocals: Record<string, any> }) {
+  return html`
+    <!doctype html>
+    <html lang="en">
+      <head>
+        ${renderEjs(import.meta.url, "<%- include('../partials/head'); %>", resLocals)}
+        ${resLocals.assessment.type === 'Exam'
+          ? html`${compiledScriptTag('examTimeLimitCountdown.ts')}
+            ${EncodedData(
+              {
+                serverRemainingMS: resLocals.assessment_instance_remaining_ms,
+                serverTimeLimitMS: resLocals.assessment_instance_time_limit_ms,
+                serverUpdateURL:
+                  resLocals.urlPrefix +
+                  '/assessment_instance/' +
+                  resLocals.assessment_instance.id +
+                  '/time_remaining',
+                canTriggerFinish: resLocals.authz_result.authorized_edit,
+                csrfToken: resLocals.__csrf_token,
+              },
+              'time-limit-data',
+            )}`
+          : ''}
+        ${compiledScriptTag('studentAssessmentInstanceClient.ts')}
+      </head>
+      <body>
+        ${resLocals.assessment.type === 'Exam' && resLocals.authz_result.authorized_edit
+          ? ConfirmFinishModal({
+              assessment_instance: resLocals.assessment_instance,
+              csrfToken: resLocals.__csrf_token,
+            })
+          : ''}
+        ${renderEjs(import.meta.url, "<%- include('../partials/navbar'); %>", {
+          ...resLocals,
+          navPage: 'assessment_instance',
+        })}
+        <main id="content" class="container">
+          <div class="card mb-4">
+            <div class="card-header bg-primary text-white">
+              ${resLocals.assessment_set.abbreviation}${resLocals.assessment.number}:
+              ${resLocals.assessment.title}
+              ${resLocals.assessment.group_work ? html`<i class="fas fa-users"></i>` : ''}
+            </div>
+
+            <div class="card-body">
+              ${resLocals.assessment.allow_real_time_grading && resLocals.assessment_instance.open
```

Comment:
```suggestion
              ${!resLocals.assessment.allow_real_time_grading && resLocals.assessment_instance.open
```

Reply:
Good catch, I don't know how I managed to lose so many `!` during the conversion. Fixed in 49bdc36.